src/crypto/tls/ech_provider.go

```diff
@@ -94,37 +90,24 @@ type EXP_ECHKeySet struct {
 	configs []byte
 
 	// Maps a configuration identifier to its secret key.
-	sk map[[maxConfigIdLen + 1]byte]EXP_ECHKey
+	sk map[uint8]EXP_ECHKey
 }
 
 // EXP_NewECHKeySet constructs an EXP_ECHKeySet.
 func EXP_NewECHKeySet(keys []EXP_ECHKey) (*EXP_ECHKeySet, error) {
+	if len(keys) > 255 {
+		return nil, fmt.Errorf("tls: ech provider:: unable to support more than 255 ECH configurations at once")
+	}
+
 	keySet := new(EXP_ECHKeySet)
-	keySet.sk = make(map[[maxConfigIdLen + 1]byte]EXP_ECHKey)
+	keySet.sk = make(map[uint8]EXP_ECHKey)
 	configs := make([]byte, 0)
 	for _, key := range keys {
-		// Compute the set of KDF algorithms supported by this configuration.
-		kdfIds := make(map[uint16]bool)
-		for _, suite := range key.config.suites {
-			kdfIds[suite.kdfId] = true
-		}
-
-		// Compute the configuration identifier for each KDF.
-		for kdfId, _ := range kdfIds {
-			configId, err := key.config.id(hpke.KDF(kdfId))
-			if err != nil {
-				return nil, err
-			}
-
-			var b cryptobyte.Builder
-			b.AddUint8LengthPrefixed(func(b *cryptobyte.Builder) {
-				b.AddBytes(configId)
-			})
-			var id [maxConfigIdLen + 1]byte // Initialized to zero
-			copy(id[:], b.BytesOrPanic())
-			keySet.sk[id] = key
+		if _, ok := keySet.sk[key.config.configId]; ok {
+			return nil, fmt.Errorf("tls: ech provider:: ECH config conflict")
```

Comment:
Remove excess `:`, consider adding more context:
```suggestion
			return nil, fmt.Errorf("tls: ech provider: ECH config conflict for configId %d", key.config.configId)
```